packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -264,6 +264,39 @@ export const useField = <
         break;
       }
 
+      // Move selection to previous or next section
+      case event.key === 'Tab': {
```

Comment:
This does seem like a sensible improvement, but given the existing approach and resulting a11y behaviour, I'd say that adding a custom tabulation management further exacerbates the a11y issues we have with the field component.
If we had spinbutton group solution, then maybe screen reader users would expect such behaviour, but now... I feel that it might even make the situation worse. 🙈 
Not to mention, that if we'd go for spinbutton solution, this extra code would basically become redundant. 🤔